### PR TITLE
Remove Chromium leftovers from sync types screen

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/17580

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Launch current Android Nightly
2. Create a sync chain
3. Open `Data Preferences`
4. Ensure there are no `Sign out ... ` and `Advanced` UI items

### CODE REVIEW

File: build/android/bytecode/java/org/brave/bytecode/BraveManageSyncSettingsClassAdapter.java [hunk 1]
```diff
@@ -16,6 +16,9 @@
     BraveManageSyncSettingsClassAdapter(ClassVisitor visitor) {
         super(visitor);
 
+        deleteField(sBraveManageSyncSettingsClassName, "mTurnOffSync");
+        makeProtectedField(sManageSyncSettingsClassName, "mTurnOffSync");
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveManageSyncSettingsClassAdapter.java:20]: Could you also add unit test to BytecodeTest.java to check that this variable still exist in parent class?

Comment (AlexeyBarabash) [reply build/android/bytecode/java/org/brave/bytecode/BraveManageSyncSettingsClassAdapter.java:20]: Updated the test at https://github.com/brave/brave-core/pull/9799/commits/1da551807122e4ec6c5b19fea15bfbb7ad113ffd . Thanks for the help.

**Comment by AlexeyBarabash** [comment]
CI for Android is marked as UNSTABLE
because `Archive JUnit-formatted test results - (0.86 sec in self)` - is yellow without explanations
https://ci.brave.com/job/pr-brave-browser-sync_cr93_fix_17580-android/2/execution/node/187/log/

**Review by samartnik — APPROVED** [review]
lgtm

**Pull request merged**